### PR TITLE
Organize biomass

### PR DESCRIPTION
**Pull request by jessicaguo**
There is now a biomass folder within the parameters_pipeline. Here, the 01_organize_chamber_biomass.Rmd collates the multiple Excel sheets for chamber biomass and leaf area. The output is saved as an .Rdata file in the sentinel-detection repo. Chamber_effect_biomass.Rmd visualizes and analyses the differences in biomass between chambers.

### CODE REVIEW

**Review by KristinaRiemer — APPROVED** [review]
I think the `chamber_biomass.Rdata` file should not be saved here in the public repo (and it's saved in the 01 script in sentinel-detection). 

For all of the biomass data cleaning, I prefer to read in entire raw datasets, instead of subsetting them in any way, in case the underlying data changes in some way. E.g., only the first ten rows of a datset are read in, with 10 specified manually, but more data is added later on but never noticed or included because that number 10 is sort of hidden within the script. This is totally a preference, but we can talk about pros and cons of these approaches. 

Lastly, I think we should be able to delete my [plot_biomass_meas.Rmd](https://github.com/az-digitalag/model-vignettes/blob/master/BioCro/DARPA/plot_biomass_meas.Rmd), which got at some of the summarizing of these biomass data, because what you have done here is much better and more extensive. Nice work!

File: parameters_processing/biomass/01_organize_chamber_biomass.Rmd [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+                          tier == "top" ~ "high")) %>%
+  mutate(temp = case_when(temp == "22" ~ "22/22",
+                          temp == "31" ~ "31/31",
+                          temp == "36" ~ "36/36")) %>%
+  mutate(light = 430)
+```
+
+Experiment 3 was conducted on ME034V-1 in 3 chambers (157, 158, and 159) and consisted of the application of GA_1 and GA_2 at different light and temperature combinations. Chamber 157 was set to a temperature of 22, while chambers 158 and 159 were set to 31. The high tier of each chamber was set to 450 light, while the low and mid tiers were set to 250. All three treatments (GA_1, GA_2, and control) were undertaken in each light and temprature combination. 
+```{r 3rd}
+# Due to additional character rows, sheet needed to be imported in two batches
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/01_organize_chamber_biomass.Rmd:108]: This sheet was annoying. There were several sections of data that were labeled  "plants_for_Jianni" across multiple columns, which cause all columns to be read as character strings. So, I read in the first section prior to those rows, then used the type from those rows to read in the rest of the sheet.

File: parameters_processing/biomass/01_organize_chamber_biomass.Rmd [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+  mutate(exp_number = 8) %>%
+  relocate(exp_number) %>%
+  tidyr::separate(location, c("chamber", "tier", NA)) %>%
+  mutate(tier = case_when(tier == "bot" ~ "low",
+                          tier == "low" ~ "low",
+                          tier == "mid" ~ "mid",
+                          tier == "top" ~ "high"))
+```
+
+Experiment 9 was conducted on ME034V-1 and an additional 7 genotypes in 2 chambers (157 and 158), all set to a day/night temperature of 31. Light was set to 430 across all chambers and tiers.
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/01_organize_chamber_biomass.Rmd:339]: Yup, changed!

File: parameters_processing/biomass/01_organize_chamber_biomass.Rmd [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+                          tier == "top" ~ "high")) %>%
+  mutate(temp = case_when(temp == "22" ~ "22/22",
+                          temp == "31" ~ "31/31",
+                          temp == "36" ~ "36/36")) %>%
+  mutate(light = case_when(light == "450" ~ 430,
+                           light == "250" ~ 250,
+                           light == "150" ~ 150))
+```
+
+Experiment 2 was conducted on ME034V-1 and mutants Les1 and Les5 in 3 chambers (157, 158, and 159). All chambers were set to temperatures of 34. In each chamber, biomass samples of Les1 were taken from low and mid tiers, Les 5 was taken from low and high tiers, and ME034V-1 was taken from all three tiers. Light is currently unknown.
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/01_organize_chamber_biomass.Rmd:72]: Updated.

File: parameters_processing/biomass/01_organize_chamber_biomass.Rmd [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+  mutate(temp = case_when(temp == "22" ~ "22/22",
+                          temp == "31" ~ "31/31",
+                          temp == "36" ~ "36/36",
+                          temp == "17" ~ "17/17"))
+```
+
+## Organization
+
+Different sheets have different column names. Below, we track which column names are present in which experiments. 
+```{r}
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/01_organize_chamber_biomass.Rmd:382]: Thanks! Still a PIA.

File: parameters_processing/biomass/02_anova_chamber_biomass.Rmd [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+library(ggplot2)
+library(readxl)
+```
+
+### High night temperature
+
+First, we focus on the high night temperature treatment, genotype ME034V-1 at 250 uE light intensity and 31/31 day/night temperature. 
+
+```{r}
+load("chamber_biomass.Rdata")
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/02_anova_chamber_biomass.Rmd:30]: Oops, I didn't realize this was still here. Please ignore, this is updated and renamed in a future PR.

File: parameters_processing/biomass/02_anova_chamber_biomass.Rmd [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+m4 <- lm(abg_biomass_DW_mg ~ temp + chamber + tier, data = subset(expdf, light == "430"))
+m5 <- lm(abg_biomass_DW_mg ~ temp + chamber, data = subset(expdf, light == "430"))
+m6 <- lm(abg_biomass_DW_mg ~ temp, data = subset(expdf, light == "430"))
+anova(m5, m6)
+anova(m4, m5)
+
+ggplot(subset(expdf, light == "430")) +
+  geom_point(aes(x = temp, y = abg_biomass_DW_mg, col = chamber, shape = tier))
+
+foo <- subset(expdf, light == "430" & temp == "31/31")
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/02_anova_chamber_biomass.Rmd:122]: Same. I *think* this is fixed in the new version.

File: parameters_processing/biomass/02_anova_chamber_biomass.Rmd [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+ggplot(foo, aes(x = chamber, y = abg_biomass_DW_mg, col = as.factor(exp_number))) +
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/02_anova_chamber_biomass.Rmd:123]: Same.

File: parameters_processing/biomass/02_anova_chamber_biomass.Rmd [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  geom_point()
+
+foo <- subset(expdf, light == "250" & temp == "31/31")
+ggplot(foo, aes(x = chamber, y = abg_biomass_DW_mg, col = as.factor(exp_number))) +
```
[the remark this answers is not in the document]
Comment (jessicaguo) [reply parameters_processing/biomass/02_anova_chamber_biomass.Rmd:127]: Same.

**Comment by jessicaguo** [comment]
Thanks for the careful review! I'm sorry to have made you review the 02_anova file, that was later renamed and reworked.

**Pull request merged**